buildSrc/src/main/java/org/opensearch/gradle/info/GlobalBuildInfoPlugin.java

```diff
@@ -195,7 +198,29 @@ private JavaVersion determineJavaVersion(String description, File javaHome, Java
     }
 
     private JvmInstallationMetadata getJavaInstallation(File javaHome) {
-        return jvmMetadataDetector.getMetadata(javaHome);
+        final InstallationLocation location = new InstallationLocation(javaHome, "Java home");
+
+        try {
+            try {
+                // The getMetadata(File) is used by Gradle pre-7.6
```

Comment:
Is it simpler to get the version of gradle and remove all these try/catches?

Reply:
Sure, but how Gradle version helps? We still need to call the method :-) Unless we set the minimum to `7.6` but in this case, it would be a "breaking" change for plugin developers - they would have to upgrade to Gradle `7.6` as well (similar pain of migration to 7.5)?

Reply:
Hm ... the old refection mechanism could be used in such a way (get methods, try to find the one you need) but `MethodHandles`'s methods do raise `NoSuchMethodException` sadly